website/source/docs/guides/agent-encryption.html.md

```diff
@@ -0,0 +1,205 @@
+
+Note: all nodes within a cluster must share the same encryption key in order to send and receive cluster information, including clients and servers. Additionally, if you're using multiple WAN joined datacenters, be sure to use _the same encryption key_ in all datacenters.
+
+### Enable Gossip Encryption: Existing Cluster
+
+Gossip encryption can also be enabled on an existing cluster, but requires several extra steps. The additional configuration of the agent configuration parameters, [`encrypt_verify_incoming`](/docs/agent/options.html#encrypt_verify_incoming) and [`encrypt_verify_outgoing`](/docs/agent/options.html#encrypt_verify_outgoing) is necessary. 
+
+**Step 1**: Generate an encryption key using `consul keygen`.
+
+```sh
+$ consul keygen
+JY34uTPZyfUE+6tinMYEVw==
+```
+
+**Step 2**: Set the [`encrypt`](/docs/agent/options.html#_encrypt) key, and set `encrypt_verify_incoming` and `encrypt_verify_outgoing` to `false` in the agent configuration file. Then initiate a rolling update of the cluster with these new values. After this step, the agents will be able to decrypt gossip but will not yet be sending encrypted traffic.
+
+```javascript
+{
+  "data_dir": "/opt/consul",
+  "log_level": "INFO",
+  "node_name": "bulldog",
+  "server": true,
+  "encrypt": "JY34uTPZyfUE+6tinMYEVw==",
+  "encrypt_verify_incoming": false,
+  "encrypt_verify_outgoing": false
+}
+```
+
+A rolling update can be made by reloading the Consul process, `consul reload` or `kill -HUP <process_id>`.
+
+**Step 3**: Update the `encrypt_verify_outgoing` setting to `true` and perform another rolling update of the cluster. The agents will now be sending encrypted gossip but will still allow incoming unencrypted traffic.
+
+```javascript
+{
+  "data_dir": "/opt/consul",
+  "log_level": "INFO",
+  "node_name": "bulldog",
+  "server": true,
+  "encrypt": "JY34uTPZyfUE+6tinMYEVw==",
+  "encrypt_verify_incoming": false,
+  "encrypt_verify_outgoing": true
+}
+```
+
+Again, use `consul reload` or `kill -HUP <process_id>` to reload the process. 
+
+**Step 4**: The previous step, enabling verify outgoing, must be completed on all agents before continuing. Update the `encrypt_verify_incoming` setting to `true` and perform a final rolling update of the cluster. 
+
+```javascript
+{
+  "data_dir": "/opt/consul",
+  "log_level": "INFO",
+  "node_name": "bulldog",
+  "server": true,
+  "encrypt": "JY34uTPZyfUE+6tinMYEVw==",
+  "encrypt_verify_incoming": true,
+  "encrypt_verify_outgoing": true
+}
+```
+
+All the agents will now be strictly enforcing encrypted gossip.
+
+## TLS Encryption for RPC
+
+Consul supports using TLS to verify the authenticity of servers and clients. To enable TLS,
+Consul requires that all clients and servers have key pairs that are generated by a single
+Certificate Authority. The CA then signs keys for each of the agents, as in
+[this tutorial on generating both a CA and signing keys](/docs/guides/creating-certificates.html)
+using [cfssl](https://cfssl.org/).
```

Comment:
Suggest we remove this since Hans is about to land #5078 which changes it so use the built-in guide.

Reply:
Removed the `cfssl` part.